packages/components/button/src/Button.tsx

```diff
@@ -0,0 +1,115 @@
+import React, { ElementType, HTMLProps } from 'react';
+import { cx } from 'emotion';
+import { Flex, CommonProps } from '@contentful/f36-core';
+import { ChevronDown } from '@contentful/f36-icons';
+import { Icon, IconComponent } from '@contentful/f36-icon';
+
+import type { ButtonVariant, ButtonSize } from './types';
+import { styles } from './styles';
+import { Spinner } from '@contentful/f36-spinner';
+
+export interface ButtonProps
+  extends Omit<HTMLProps<HTMLButtonElement & HTMLAnchorElement>, 'size'>,
+    CommonProps {
+  children?: React.ReactNode;
+  /**
+   * Determines style variation of Button component
+   * @default secondary
+   */
+  variant?: ButtonVariant;
+  /**
+   * Determines size variation of Button component
+   * @default medium
+   */
+  size?: ButtonSize;
+  /**
+   * Applies active styles
+   * @default false
+   */
+  isActive?: boolean;
+  /**
+   * Disabled interaction and applies disabled styles
+   * @default false
+   */
+  disabled?: boolean;
+  /**
+   * Button html type attribute
+   */
+  type?: 'submit' | 'button' | 'reset';
+  /**
+   * Adds dropdown indicator icon
+   */
+  isDropdown?: boolean;
+  /**
+   * Expects any of the icon components
+   */
+  icon?: IconComponent;
+  /**
+   * Adds loading indicator icon and disables interactions
+   */
+  isLoading?: boolean;
+  /**
+   * Forces button to take 100% of the container
+   */
+  isFullWidth?: boolean;
+}
+
+const _Button = (props: ButtonProps, ref) => {
+  const {
+    children,
+    variant = 'secondary',
+    size = 'medium',
+    href,
+    disabled,
+    type = !href ? 'button' : undefined,
+    icon,
+    isActive,
+    isDropdown,
+    isLoading,
+    isFullWidth,
+    ...otherProps
+  } = props;
+
+  const rootClassNames = cx(
+    styles.button({
+      variant,
+      size,
+      isActive,
+      isDisabled: disabled,
+      isFullWidth,
+    }),
+  );
+
+  const Element: ElementType = href ? 'a' : 'button';
+
+  return (
+    <Element
```

Comment:
Should we use `Box` or similar here to get the different margin props?

Reply:
hmm, I'm not sure, `Box` seems like an overhead to add another layer/component as a base for a simple button just for margin props, if we want margins maybe we could add a prop just for that

Reply:
Could just extend `MarginProps` but I don't think using `Box` is a problem as long as we use it. Maybe here it's best not to since we don't want the `PaddingProps` as well.

Reply:
Cool, are we sure about the `margin` props or should we discuss it in an issue first? I barely add margins to buttons, so I'm not sure if it's worth the code

Reply:
Okay let's take it to a separate issue 🙂